### PR TITLE
Add HibernateExceptionMappers.kt

### PR DESCRIPTION
**Pull request by jjestrel**
map a bunch of Hibernate conflict exceptions to 409s

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import misk.web.toResponseBody
+import okhttp3.Headers
+import org.hibernate.StaleObjectStateException
+import org.hibernate.exception.ConstraintViolationException
+import java.sql.SQLIntegrityConstraintViolationException
+import javax.inject.Inject
+import javax.persistence.OptimisticLockException
+
+internal class RetryTransactionExceptionMapper @Inject internal constructor(
+  val config: ActionExceptionLogLevelConfig
```
Comment (jjestrel) [inline misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt]: We copy this from `ActionExceptionMapper`. maybe this should be renamed?

File: misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+internal class OptimisticLockExceptionMapper @Inject internal constructor(
+) : ExceptionMapper<OptimisticLockException> {
+  override fun toResponse(th: OptimisticLockException): Response<ResponseBody>
+    = ConflictExceptionResponder.toResponse()
+
+  override fun canHandle(th: Throwable): Boolean = th is OptimisticLockException
+
+  override fun loggingLevel(th: OptimisticLockException) = Level.WARN
```
Comment (jjestrel) [inline misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt]: This matches `ActionExceptionMapper`'s default

File: misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+object ConflictExceptionResponder {
```
Comment (swankjesse) [inline misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt]: internal?

Comment (jjestrel) [reply misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt]: changed

File: misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import misk.web.toResponseBody
+import okhttp3.Headers
+import org.hibernate.StaleObjectStateException
+import org.hibernate.exception.ConstraintViolationException
+import org.slf4j.event.Level
+import java.sql.SQLIntegrityConstraintViolationException
+import javax.inject.Inject
+import javax.persistence.OptimisticLockException
+
+internal class RetryTransactionExceptionMapper @Inject internal constructor(
```
Comment (ryanhall07) [inline misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt]: this does not seem like a general mapper for `RetryTransactionException`, but very specific to the actual cause of the root exception. probably needs a better name to make that clear

Comment (jjestrel) [reply misk-hibernate/src/main/kotlin/misk/hibernate/HibernateExceptionMappers.kt]: agreed it's not general. when we have another case I'll rewrite this to get the root cause and check if there's a mapper for it. Or maybe `Throwables.getCasualChains()` and keep going down until there's an exception mapper (or up?)

**Pull request merged**